library/system/src/lib/yast2/etc_fstab.rb

```diff
@@ -0,0 +1,566 @@
+        # and truly fixed), we can avoid that endless loop.
+        #
+        # The fstab is of course still incorrect, but there is nothing we can
+        # do about that at this point.
+      else
+        # Take this entry out of the entries and put it back at the correct
+        # place.
+        @entries.delete_at(problem_index)
+        add_entry(entry)
+
+        # Keep track of the reordered entries to avoid an endless loop
+        reordered << entry
+      end
+    end
+    success
+  end
+
+  # Find the the entry index of the next mount order problem starting from
+  # 'start_index' or -1 if there is no more.
+  #
+  # @param start_index [Fixnum]
+  #
+  # @return [Fixnum] Next problematic entry index or -1 if no more problems
+  #
+  def next_mount_order_problem(start_index = 0)
+    each_with_index do |entry, index|
+      next if index < start_index
+      sort_index = find_sort_index(entry)
+      next if sort_index == -1
+      return index if sort_index < index
+    end
+    -1
+  end
+
+protected
+
+  # Find the correct index for an entry if it depends on any other entry or -1
+  # if no other entry depends on this one, i.e. it can safely added to the end
+  # of the list.
+  #
+  # @param new_entry [EtcFstab::Entry]
+  #
+  # @return [Fixnum] correct index or -1 if there is no dependency
+  #
+  def find_sort_index(new_entry)
+    mount_point = new_entry.mount_point
+    each_with_index do |entry, index|
+      next if entry.equal?(new_entry)
+      return index if entry.mount_point.start_with?(mount_point)
+    end
+    -1
+  end
+
+public
+
+  # Get the "mount by" type of a device entry in /etc/fstab.
+  # See also EtcFstab::Entry.get_mount_by.
+  #
+  # @param device [String] device field or complete entry line
+  # @return [Symbol] One of :uuid, :label, :id, :path, :device
+  #
+  def self.get_mount_by(device)
+    case device
+    when /^UUID=/, %r{^/dev/disk/by-uuid/}
+      :uuid
+    when /^LABEL=/, %r{^/dev/disk/by-label/}
+      :label
+    when %r{^/dev/disk/by-id/}
+      :id
+    when %r{^/dev/disk/by-path/}
+      :path
+    else
+      :device
+    end
+  end
+
+  # Encode an fstab entry. It may sound surprising, but the file format
+  # actually allows space characters in certain places, such as the mount
+  # point; the reasoning is that a space character is a permitted character in
+  # a directory name, so there has to be a way to specify such a path without
+  # breaking the file format. According to "man fstab", this is done by
+  # encoding the space character in octal, i.e. as \040. This is the function
+  # to do it.
+  #
+  # This class and the corresponding entry class handles this transparently, so
+  # this should never be necessary to use from the outside.
+  #
+  # @param unencoded [String] String with possible space characters
+  # @return [String] String with space characters replaced by \040
+  #
+  def self.fstab_encode(unencoded)
+    unencoded.gsub(" ", '\\\\040')
```

Comment:
I don't think this will work: `String::tr()` translates each character that appears in the first string to exactly one from the second string; AFAICS you cannot replace one from the first string to several from the second. What /etc/fstab needs here, however, is exactly that: Each blank is to be replaced by the sequence `\040`, i.e. a literal backslash followed by three digits specifying the octal code.

Reply:
Oh, good point, I did not realize that the replacement is longer...